script.js

```diff
@@ -0,0 +1,167 @@
+        dropdownItem.setAttribute('data-region', region);
+        dropdownItem.textContent = region;
+        dropdownList.appendChild(dropdownItem);
+    });
+    dropdownRegionFilter.appendChild(dropdownList);
+}
+
+function getSearchdata() {
+    const searchInput = document.querySelector('.search-input');
+    searchInput.addEventListener('input', (e) => {
+        searchValue = e.target.value;
+        const filteredData = filterDataSearch(countryFetchedData, searchValue);
+        countriesGrid.innerHTML = '';
+        analyzeMainContainer(filteredData);
+    });
+}
+
+function filterDataSearch(data, searchValue) {
+    return data.filter(({ name }) => name.toLowerCase().includes(searchValue.toLowerCase()));
+}
+
+function analyzeMainContainer(data) {
+    data.forEach(({ name, flag, population, region, capital }) => {
+        const countryCard = document.createElement('a');
+        countryCard.href = './details.html';
+        countryCard.addEventListener('click', function () {
+            const countryData = { name, flag, population, region, capital };
+            localStorage.setItem('countryDetails', JSON.stringify(countryData));
+        });
+
+        countryCard.className = 'country scale-effect';
+        countryCard.setAttribute('data-country-name', name);
+
+        countryCard.innerHTML = `
+      <div class="country-flag">
+        <img src="${flag}" alt="${name} Flag">
+      </div>
+      <div class="country-info">
+        <h2 class="country-title">${name}</h2>
+        <ul class="country-brief">
+          <li><strong>Population: </strong>${population}</li>
+          <li><strong>Region: </strong>${region}</li>
+          <li><strong>Capital: </strong>${capital}</li>
+        </ul>
+      </div>`;
+
+        countriesGrid.appendChild(countryCard);
+    });
+}
+
+function handleDropdownClick() {
+    const dropdownWrapper = document.querySelector('.dropdown-wrapper');
+    dropdownWrapper.addEventListener('click', function () {
+        this.classList.toggle('open');
+    });
+
+    const dropdownItems = dropdownWrapper.querySelector('.dropdown-body ul');
+    dropdownItems.addEventListener('click', function (e) {
+        e.stopPropagation();
+        const selectedRegion = e.target.getAttribute('data-region');
+        if (selectedRegion) {
+            dropdownWrapper.classList.remove('open');
+            getCountriesByRegion(selectedRegion);
+        }
+    });
+}
+
+function getCountriesByRegion(region) {
+    let selectedDropdoenOrigin;
+    if (region === 'all') {
+        selectedDropdoenOrigin = countryFetchedData;
+    } else {
+        selectedDropdoenOrigin = countryFetchedData.filter(country => country.region === region);
+    }
+    countriesGrid.innerHTML = '';
+    analyzeMainContainer(selectedDropdoenOrigin);
+}
+
+async function displayData() {
+    await fetchCountriesData();
+    if (countryFetchedData) {
+        analyzeMainContainer(countryFetchedData);
```

Comment:
You can use [guard clause](https://medium.com/@timothydan/javascript-guard-clauses-64b999e3240) for early return making the code cleaner : 

```js
async function displayData() {
    await fetchCountriesData();

    if (!countryFetchedData) {
        console.error('Error displaying countries data');
        return;
    }

    analyzeMainContainer(countryFetchedData);
    getDropdownsRegions(countryFetchedData);
    getSearchdata();
    handleDropdownClick();
}
```